10.0 migration/l10n br account

renatonlima opened the pull request.

yk2kus committed:
migrate l10n_br_account to V 10.0

renatonlima committed:
Updated module version and remove unused dependency

renatonlima committed:
Removed Workflows

renatonlima committed:
[FIX] imports

renatonlima committed:
Migrated object account.tax

renatonlima committed:
Removed obsolete class AccountAccount

renatonlima committed:
Migrate l10n_br_account objects

renatonlima committed:
migrate account.invoice and account.invoice.line

renatonlima committed:
migrate fiscal position

renatonlima committed:
migrate fiscal position rule

renatonlima committed:
removed unused demo data

renatonlima committed:
migrate base demo data

renatonlima committed:
migrate l10n_br_account data

renatonlima committed:
migrate security files

renatonlima committed:
migrate invoice report

renatonlima committed:
migrate fiscal position rule view

renatonlima committed:
migrate account invoice view

renatonlima committed:
migrate account view

renatonlima committed:
migrate account fiscal position view

renatonlima committed:
migrate res.company object

renatonlima committed:
remove object l10n_br_account.product.category from product.py

renatonlima committed:
added object l10n_br_account_product_category in new file l10n_br_account_product_category.py

renatonlima committed:
added new py file in __init__.py

renatonlima committed:
removed account.fiscal.position objects from res.partner.py

renatonlima committed:
added account.fiscal.position objects files

renatonlima committed:
addded new files in __init__

renatonlima committed:
remove account.journal and wizard_multi_charts_accounts objects from account.py file

renatonlima committed:
added account_journal.py file

renatonlima committed:
wizard.multi.charts.accounts file

renatonlima committed:
added new files in __init__

renatonlima committed:
change name file

renatonlima committed:
move account_invoice_line to new file

renatonlima committed:
migrate and change file name product.py to product_template.py

renatonlima committed:
migrate product.template object

renatonlima committed:
remove object from l10n_br_account.py

renatonlima committed:
add new file for l10n_br_account.cnae object

renatonlima committed:
add new file for l10n_br_account.fiscal.category object

renatonlima committed:
add new file for l10n_br_account.fiscal.document object

renatonlima committed:
add new file for l10n_br_account.fiscal.document.serie object

renatonlima committed:
add new file for l10n_br_account.fiscal.partner.type object

renatonlima committed:
add new file for l10n_br_account.service.type object

renatonlima committed:
add new file for l10n_br_account.product.category object

renatonlima committed:
add new file for l10n_br_account.tax.definition object

renatonlima committed:
update __init__ file

renatonlima committed:
pep8

renatonlima committed:
[FIX] worng import

sadamo commented on l10n_br_account/l10n_br_account_workflow.xml:
```diff
@@ -1,79 +0,0 @@
-    <data>
-
-        <!-- workflow -->
-        <record id="l10n_br_account_fiscal_category_workflow" model="workflow">
```
Essas transições de Workflow foram retiradas e assim as Categorias Fiscais não funcionam corretamente. É necessário ou criar métodos novos para os botões ou voltar com o workflow

sadamo commented on l10n_br_account/account_fiscal_position_workflow.xml:
```diff
@@ -1,79 +0,0 @@
-    <data>
-
-        <!-- workflow -->
-        <record id="l10n_br_account_fiscal_position_workflow" model="workflow">
```
As Posições Fiscais ainda contém o campo de state, ele é utilizado em algo?

sadamo commented on l10n_br_account/models/account_invoice.py:
```diff
@@ -119,22 +160,57 @@ def finalize_invoice_move_lines(self, move_lines):
+
+    @api.model
+    def invoice_line_move_line_get(self):
+        result = super(AccountInvoice, self).invoice_line_move_line_get()
```
Essa aplicação do desconto no valor do lançamento não causa um desencontro de informações para saber o pq do valor ter mudado? Essa é a melhor forma de aplicar o desconto?

sadamo reviewed (changes requested):
Alguns pontos para rever:
* Constraint no cancelamento de invoice utilizando api antiga
* Visão da Regra de Posição Fiscal com campos desalinhados
* Constraint na inutilização de faixa sendo aplicada 2 vezes? (api antiga e nova)
* Constraint da categoria fiscal precisa ser SQL? Acredito que uma constraint utilizando a api nova já seria suficiente, além de caso seja necessária alguma mudança, não precisa alterar o banco (retirar a constraint no banco)
* Várias utilizações do decorator api.one
* Objeto account.tax não possui mais o campo "base_reduction" que está sendo utilizado no código
* métodos compute_display_name() com api.multi e ensure_one().  "for record in self:" seria uma abordagem melhor?

BiancaBartolomei commented:
No modelo l10n_br_account, tem o "TODO" para passar o nome dos campos para inglês.                          l10n_br_account/tests está vazio, então poderia ser retirado.

mbcosta committed:
Replace api.one to api.multi .

mbcosta committed:
Change tests from YAML to Python.

mbcosta committed:
PEP8.

mbcosta committed:
[IMP] Tests.

mbcosta committed:
[REF] The code should be implement on the module l10n_br_account_product .

mbcosta committed:
[FIX] The method 'execute' from object wizard.multi.charts.accounts check field 'cfop_id' when create template.

mbcosta committed:
[IMP] Tests.

renatonlima committed:
[MIGR] view file product_view.xml

renatonlima committed:
[MIGR] view file res_company_view.xml

renatonlima committed:
[MIGR] view file res_config_view.xml

mbcosta committed:
[REF] Move wizard to create charts to module l10n_br_account_product because CFOP is implement there.

mbcosta committed:
[REF]  Moved method to module l10n_br_account_product because the implement of field CFOP is there.

mbcosta committed:
[IMP] Tests.

mbcosta committed:
PEP8.

mbcosta committed:
[REF] The field 'move_line_receivable_id' are changed to 'payment_move_line_ids' on core.

mbcosta committed:
[IMP] Tests.

mbcosta committed:
PEP8.

mbcosta committed:
[REF] The function should be in l10n_br_account_product module.

mbcosta committed:
[IMP] Tests.

mileo commented on .travis.yml:
```diff
@@ -16,8 +16,7 @@ addons:
 env:
   - VERSION="10.0" LINT_CHECK="1"
-  - VERSION="10.0" ODOO_REPO="odoo/odoo" LINT_CHECK="0"
-  - VERSION="10.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0"
```
Isto deve corrigido posteriormente

mileo commented on l10n_br_account/models/account_fiscal_position.py:
```diff
@@ -0,0 +1,52 @@
+)
+
+
+class AccountFiscalPosition(AccountFiscalPositionAbstract,
```
Esta tela não esta funcionando corretamente, não foi possível inserir dados no campo tax_ids e o teste automatizado esta viciado, pois não passa pelo método map_tax que utiliza os campos deste modelo.

mbcosta replied:
@mileo aqui tem um problema que vi com o @renatonlima tanto o modulo l10n_br_account_product quanto o l10n_br_account_service vão usar esse método mas não é usado aqui ( ou podemos ver uma forma de chama-lo ) se for removido daqui vai ser preciso duplicar o código nesses módulos

renatonlima replied:
@mileo para resolver este questão eu previso pegar o onchange_fiscal() que esta no módulo l10n_br_account_product e coloca-lo no l10n_br_account. para eu devo fazer esta mudança depois deste merge para não ter que fazer o rebase na migração l10n_br_account_product, pois vai dar muito conflito fazer isso.

mileo commented on l10n_br_account/models/account_invoice.py:
```diff
@@ -22,83 +21,134 @@
-        'account.move.line', string='Receivables',
+        comodel_name='account.move.line',
+        string=u'Receivables',
         compute='_compute_receivables')
```
Este método foi removido, mas seu compute esta sendo chamado aqui. Impossibilitando a criação de invoices.

```
Traceback (most recent call last):
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/http.py", line 642, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/http.py", line 684, in dispatch
    result = self._call_function(**self.params)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/http.py", line 334, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/http.py", line 327, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/http.py", line 942, in __call__
    return self.method(*args, **kw)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/http.py", line 507, in response_wrap
    response = f(*args, **kw)
  File "/home/mileo/Projects/oca10/parts/odoo/addons/web/controllers/main.py", line 895, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/mileo/Projects/oca10/parts/odoo/addons/web/controllers/main.py", line 887, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/models.py", line 3079, in read
    values[name] = field.convert_to_read(record[name], record, use_name_get)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/models.py", line 5288, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/fields.py", line 942, in __get__
    self.determine_value(record)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/fields.py", line 1054, in determine_value
    self.compute_value(recs)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/fields.py", line 1008, in compute_value
    self._compute_value(records)
  File "/home/mileo/Projects/oca10/parts/odoo/odoo/fields.py", line 999, in _compute_value
    getattr(records, self.compute)()
AttributeError: 'account.invoice' object has no attribute '_compute_receivables'
```

Foram realizados testes funcionais?

mbcosta replied:
O campo move_line_receivable_id foi removido do "core" e portanto a função também deveria '_compute_receivables'  ou teremos que implementa-lo na localização. Sobre os testes funcionais esse modulo na verdade é um modulo intermediário que será utilizado pelo l10n_br_account_product e l10n_br_account_service é existe para que não seja duplicado código nesses módulos, isso é tudo que for comum entre eles deveria estar aqui no l10n_br_account, eu também questionei isso com o @renatonlima  , se você acredita ser necessário o modulo criar menus ou telas para permitir testes funcionais podemos verificar quais seriam

mileo reviewed (changes requested):
Não foi possível realizar o teste do PR. 

Favor verificar os comentários no código. 

Somente depois de corrigido será possível realizar a validação do PR.

mileo committed:
[MIG] Upport das permissões de segurança da v8

3a875b36c20487ba75c260dc30b5670096a25daf

f6dcd899cb005269eafaacc2eb14f8ef39cbb670

30d6fa66189718dc6fc4f92dc63854b94ca8fb7a

7276ca32bfa4fa26f877f449fdf65abb4c71b505

57e8d702d76e3c41086d656820749b65e32d97a5

78694e93b533314e3bf3c4b3b18f8636de0c8343

mileo committed:
[MIG] _compute_receivables of account.invoice

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mbcosta commented on l10n_br_account/wizard/account_invoice_confirm.py:
```diff
@@ -0,0 +1,25 @@
+                    _('Warning!'),
+                    _("Selected invoice(s) cannot be confirmed as they are "
+                      "not in 'Draft' or 'Pro-Forma' state."))
+            record.signal_workflow('invoice_validate')
```
Workflow foi removido na versão 10.0

mileo replied:
Foi removido na versão 11, na 10 ainda esta presente.

https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/workflow/workflow.py

mileo replied:
Agora entendi o que vc falou, o workflow da invoice foi removido na versão 10, apesar de ainda presente no core.

mbcosta commented on l10n_br_account/models/account_invoice.py:
```diff
@@ -52,33 +52,34 @@ def _compute_amount(self):
 
-    @api.one
     @api.depends('move_id.line_ids.amount_residual')
     def _compute_payments(self):
```
@mileo essa função esta vindo do "core" por isso foi usado o api.one, podemos ter problemas se a função que esta sendo herdada alterar isso.

mbcosta commented on l10n_br_account/models/account_invoice.py:
```diff
@@ -67,6 +67,19 @@ def _compute_payments(self):
+                line.account_id.internal_type in ('receivable', 'payable') and \
+                    self.journal_id.revenue_expense:
+                lines |= line
+        self.move_line_receivable_id = lines.sorted()
```
o campo move_line_receivable_id não existe mais, será implementado na localização ?

mileo committed:
[FIX] Convert invoice computed methods to api multi

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mbcosta committed:
[FIX] There is a field in Tax object to identify the kind of tax(ex.: percent, fixed, etc).

mbcosta committed:
[FIX] Logical problem in function _compute_receivables .

mbcosta committed:
[IMP] Tests.

rvalyi reviewed (approved):
OK for merging. I have a few docstring remarks but let's do that later. Also let's add some more tests once this is merged and l10n_br_account_product is rebased on it.

renatonlima committed:
comment method

renatonlima committed:
pep8
